### PR TITLE
Re-arrange tests

### PR DESCRIPTION
**Pull request by snopoke**
I'm finding some of the test files in app builder are becoming unwieldy so decided to start splitting some of them out by 'feature'.

This is pretty much all just moving stuff around except for the last commit where I updated a test to remove its dependency on an app XML file (which it should not have been using in the first place).

@TylerSheffels  

cc @proteusvacuum I know you've done the same with the test code for scheduler but also nice to group the test fixtures into their own folder (if there are enough to make a difference)

### CODE REVIEW

File: corehq/apps/app_manager/tests/test_case_list_form.py [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+        self.app = Application.new_app('domain', 'New App', APP_V2)
+        self.app.version = 3
+
+    def tearDown(self):
+        self.is_usercase_in_use_patch.stop()
+
+    def _add_module_and_form(self, ModuleClass):
+        self.module = self.app.add_module(ModuleClass.new_module('New Module', lang='en'))
+        self.module.case_type = 'test_case_type'
+        self.form = self.module.new_form("Untitled Form", "en", self.get_xml('original_form', override_path=('data',)))
```
Comment (dimagimon) [inline corehq/apps/app_manager/tests/test_case_list_form.py]: E501 line too long (119 > 115 characters)

File: corehq/apps/app_manager/tests/test_form_preparation_v2.py [hunk 2]
```diff
@@ -254,7 +248,7 @@ def setUp(self):
         self.app.version = 3
         self.module = self.app.add_module(AdvancedModule.new_module('New Module', lang='en'))
         self.module.case_type = 'test_case_type'
-        self.form = self.module.new_form("Untitled Form", "en", self.get_xml('original'))
+        self.form = self.module.new_form("Untitled Form", "en", self.get_xml('original_form', override_path=('data',)))
```
Comment (dimagimon) [inline corehq/apps/app_manager/tests/test_form_preparation_v2.py]: E501 line too long (119 > 115 characters)

**Comment by czue** [comment]
nice

**Comment by proteusvacuum** [comment]
sweet!

**Comment by snopoke** [comment]
:white_check_mark:

**Comment by snopoke** [comment]
@TylerSheffels ping

**Comment by snopoke** [comment]
(self merging)

**Comment by TylerSheffels** [comment]
Looks great :+1:

**Pull request merged**